tests/test_ssl_transport_close.py

```diff
@@ -0,0 +1,83 @@
+from aiohttp import web
+import aiohttp
+import asyncio
+import warnings
+
+
+async def handle_root(request):
+    """
+    respond after server was restarted, forcing connection_lost
+    """
+    cq = request.app['cq']
+    dq = request.app['cq']
+    i = await cq.get()
+    cq.task_done()
+    await dq.put(i)
+    return web.Response(text='')
+
+
+async def _prepare(aiohttp_server, ssl_ctx, client_ssl_ctx, cq, dq):
+    app = web.Application()
+    app['cq'] = cq
+    app['dq'] = dq
+    app.router.add_get('/', handle_root)
+    server = await aiohttp_server(app, port=12345, ssl=ssl_ctx)
+    asyncio.ensure_future(_restart(server.runner, ssl_ctx, cq))
+    session = aiohttp.ClientSession(connector=aiohttp.TCPConnector(ssl=client_ssl_ctx))
+    return server, session
+
+
+async def _close(server, session):
+    await server.close()
+    await session.close()
+
+async def _restart(runner, ssl_ctx, cq):
+    """
+    restart service to force connection_lost
+    """
+    while True:
+        for s in list(runner.sites):
+            await s.stop()
+            s = web.TCPSite(runner, '127.0.0.1', port=12345, ssl_context=ssl_ctx)
```

Comment:
What is the point of creating a new site object on the same port?

Reply:
closing the site and re-opening the site to force a client side connection_lost